executor/builder.go

```diff
@@ -656,12 +660,74 @@ func (b *executorBuilder) buildRevoke(revoke *ast.RevokeStmt) Executor {
 }
 
 func (b *executorBuilder) buildDDL(v *plannercore.DDL) Executor {
-	e := &DDLExec{
+	if b.ctx.GetSessionVars().CreateTableInsertingID != 0 {
+		// in a 'inserting data from select' state of creating table.
```

Comment:
Could you add more comments here, to explain why we implement create select so weird? And add a TODO.

Reply:
Sure, thanks!